File: OOP/02_28/AnimalKingdomEnhanced/src/main/java/edu/cuny/brooklyn/oop/animal/Cat.java
package edu.cuny.brooklyn.oop.animal;

//public class Cat extends Feline {
//
//	public Cat(String name) {
//		super(name);
//	}
//
//	@Override
//	public void makeNoise() {
//		System.out.println("Meowing");
//	}
//
//}


public class Cat extends Feline implements CatMotion {

	public Cat(String name) {
		super(name);
	}

	@Override
	public void makeNoise() {
		System.out.println("Meowing");
	}

	public void tap(Animal animal) {
		System.out.println("Tapping " + animal.getName());
	}
}

File: OOP/03_05/cpp/cat.cc
#include <iostream>
#include "cat.h"

namespace Edu {
    namespace Brooklyn {

        Cat::Cat() {
            this->name = "cat who does not yet to have a name";
        }

        Cat::Cat(std::string name) {
            this->name = name;
        }

        void Cat::tap(const Cat& other) const {
            std::cout << this->name << " is tapping " << other.name << std::endl;
        }

        std::string Cat::getName() const {
            return name;
        }
        
    }
}
